src/fosslight_yocto/create_oss_report_for_yocto.py

```
@@ -901,8 +902,19 @@ def get_recipe_for_src_analysis(analyze_all):
         if not analyze_all:
             db_conn, db_cur = connect_to_osc_db()
             if db_conn != "" and db_cur != "":
+                names = list(set([item.get('name', '') for item in oss_list.values() if item.get('name', '')]))
+                links = list(set([item.get('link', '') for item in oss_list.values() if item.get('link', '')]))
+
+                existing_names, existing_links = check_oss_exists_in_db_batch(db_cur, names, links)
+
                 for key, oss_item in oss_list.items():
```

Ruff (0.15.10)

[warning] Loop control variable `key` not used within loop body

(B007)